redis/connection.py

```diff
@@ -1185,23 +1215,7 @@ def get_connection(self, command_name: str, *keys, **options) -> "Connection":
             self._in_use_connections.add(connection)
 
         try:
-            # ensure this connection is connected to Redis
-            connection.connect()
-            # if client caching is not enabled connections that the pool
-            # provides should be ready to send a command.
-            # if not, the connection was either returned to the
-            # pool before all data has been read or the socket has been
-            # closed. either way, reconnect and verify everything is good.
-            # (if caching enabled the connection will not always be ready
-            # to send a command because it may contain invalidation messages)
-            try:
-                if connection.can_read() and connection.client_cache is None:
-                    raise ConnectionError("Connection has data")
-            except (ConnectionError, OSError):
-                connection.disconnect()
-                connection.connect()
-                if connection.can_read():
-                    raise ConnectionError("Connection not ready")
```

Comment:
I refactored this to `ensure_connection` so that the subclass can just directly call this. There's also a similar block of code in `BlockingConnectionPool`, except that it doesn't have the `and connection.client_cache is None` clause. But it looks to me like there's nothing special with `client_cache` in `BlockingConnectionPool`
Should we refactor that out as well, or is the absence of that clause an intentional distinction for `BlockingConnectionPool`?